packages/@aws-cdk/aws-efs/lib/efs-file-system.ts

```diff
@@ -0,0 +1,296 @@
+   */
+  private static readonly DEFAULT_PORT: number = 2049;
+
+  /**
+   * The security groups/rules used to allow network connections to the file system.
+   */
+  public readonly connections: ec2.Connections;
+
+  /**
+   * Identifiers of the mount targets
+   */
+  public readonly mountTargetIdentifiers: string[] = [];
+
+  /**
+   * IP Addresses of the mount targets
+   */
+  public readonly mountTargetIpAddress: string[] = [];
+
+  /**
+   * @attribute
+   */
+  public readonly fileSystemId: string;
+
+  private readonly mountTargets: CfnMountTarget[] = [];
+  private readonly efsFileSystem: CfnFileSystem;
+
+  /**
+   * Constructor for creating a new EFS FileSystem.
+   */
+  constructor(scope: Construct, id: string, props: EfsFileSystemProps) {
+    super(scope, id);
+
+    if (props.throughputMode === EfsThroughputMode.PROVISIONED) {
+      if (props.provisionedThroughputInMibps === undefined) {
+        throw new Error('Property provisionedThroughputInMibps is required when throughputMode is PROVISIONED');
+      } else if (!Number.isInteger(props.provisionedThroughputInMibps)) {
+        throw new Error("Invalid input for provisionedThroughputInMibps");
+      } else if (props.provisionedThroughputInMibps < 1 || props.provisionedThroughputInMibps > 1024) {
+        this.node.addWarning("Valid values for throughput are 1-1024 MiB/s. You can get this limit increased by contacting AWS Support.");
+      }
+    }
+
+    this.efsFileSystem = new CfnFileSystem(this, "FileSystem", {
+      encrypted: props.encrypted,
+      kmsKeyId: (props.kmsKey ? props.kmsKey.keyId : undefined),
+      lifecyclePolicies: (props.lifecyclePolicy ? Array.of({
+        transitionToIa: EfsLifecyclePolicyProperty[props.lifecyclePolicy]
+      } as CfnFileSystem.LifecyclePolicyProperty) : undefined),
+      performanceMode: props.performanceMode,
+      throughputMode: props.throughputMode,
+      provisionedThroughputInMibps: props.provisionedThroughputInMibps
+    });
+
+    this.fileSystemId = this.efsFileSystem.ref;
+    this.node.defaultChild = this.efsFileSystem;
+
+    const securityGroup = (props.securityGroup || new ec2.SecurityGroup(this, 'EfsSecurityGroup', {
+      vpc: props.vpc
+    }));
+
+    this.connections = new ec2.Connections({
+      securityGroups: [securityGroup],
+      defaultPort: ec2.Port.tcp(EfsFileSystem.DEFAULT_PORT)
+    });
+
+    const subnets = props.vpc.selectSubnets(props.vpcSubnets);
+
+    // We now have to create the mount target for each of the mentioned subnet
+    let mountTargetCount = 0;
+    subnets.subnetIds.forEach((subnetId: string) => {
+      const efsMountTarget = new CfnMountTarget(this,
+        "EfsMountTarget" + (++mountTargetCount),
+        {
+          fileSystemId: this.fileSystemId,
+          securityGroups: Array.of(securityGroup.securityGroupId),
+          subnetId
+        });
+      this.mountTargets.push(efsMountTarget);
+      this.mountTargetIdentifiers.push(efsMountTarget.ref);
```

Comment:
I see we're now holding on to even more attributes of the mount targets, but why would you need these?

The UserData script you gave uses only the ID and nothing else. Why do we need the IDs and IP addresses?

Reply:
The file system can be mounted using the IP addresses as well. The UserData in the example creates the domain name from file system and uses it for mounting. Some users may need the subnet based on IP address to mount it, hence exposing the IP address of all the mount targets. 

Exposing Ids is not required, but I took the reference from `aws-rds` that we do expose both the identifiers and ip address of all the instances, hence exposing to the user.

Should I remove these for now?

Reply:
This is probably fine, although in practice people are going to want to use the mount targets in the same AZs as their instances (to cut on cross-AZ traffic fees).

Can the construct accomodate that in its current state?

Reply:
Oh I see, using the domain name you don't have to: https://docs.aws.amazon.com/efs/latest/ug/mounting-fs-mount-cmd-general.html

Given that I feel the IP addresses are useless without subnet information to identify them, I vote we don't expose any of the mount target information right now and wait for people to come to us with use cases where they need IP addresses. Then they can tell us how they expect to find the IP address corresponding to their AZ.

Reply:
Removing the mount targets for now.
I can think of a way for exposing the IP address along with the subnet information. Will create another PR for the same.